js/src/admin/components/ExtensionPage.js

```diff
@@ -0,0 +1,251 @@
+          </div>
+        </div>
+        {!this.isEnabled() ? (
+          <div className="container">
+            <h2 className="ExtensionPage-subHeader">{app.translator.trans('core.admin.extension.enable_to_see')}</h2>
+          </div>
+        ) : (
+          <div className="ExtensionPage-body">
+            <div className="ExtensionPage-settings">
+              <div className="container">
+                {this.form() ? (
+                  <div className="Form">
+                    {this.form()}
+
+                    <div className="Form-group">{this.submitButton()}</div>
+                  </div>
+                ) : app.extensionSettings[this.extension.id] ? (
+                  <Button onclick={app.extensionSettings[this.extension.id].bind(this)} className="Button Button--primary">
+                    {app.translator.trans('core.admin.extension.open_modal')}
+                  </Button>
+                ) : (
+                  <h2 className="ExtensionPage-subHeader">{app.translator.trans('core.admin.extension.no_settings')}</h2>
+                )}
+              </div>
+            </div>
+            <div className="ExtensionPage-permissions">
+              <div className="ExtensionPage-permissions-header">
+                <h2 className="ExtensionTitle">{app.translator.trans('core.admin.extension.permissions_title')}</h2>
+              </div>
+              <div className="container">
+                {app.extensionPermissions[this.extension.id] ? (
+                  ExtensionPermissionGrid.component({ extensionId: this.extension.id })
+                ) : (
+                  <h2 className="ExtensionPage-subHeader">{app.translator.trans('core.admin.extension.no_permissions')}</h2>
+                )}
+              </div>
+            </div>
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  form() {
+    return '';
+  }
+
+  headerItems() {
+    const items = new ItemList();
+
+    items.add(
+      'uninstall',
+      Button.component(
+        {
+          icon: 'far fa-trash-alt',
+          onclick: () => {
+            app
+              .request({
+                url: app.forum.attribute('apiUrl') + '/extensions/' + this.extension.id,
+                method: 'DELETE',
+              })
+              .then(() => window.location.reload());
+
+            app.modal.show(LoadingModal);
+          },
+        },
+        app.translator.trans('core.admin.extension.uninstall_button')
+      )
+    );
+
+    return items;
+  }
+
+  infoItems() {
+    const items = new ItemList();
+
+    if (this.extension.authors) {
+      console.log(this.extension);
+      let authors = [];
+      console.log(this.extension.authors);
+
+      Object.keys(this.extension.authors).map((author) => {
+        authors.push(this.extension.authors[author].name);
+      });
+
+      items.add('authors', [icon('fas fa-user'), <span>{authors.join(', ')}</span>]);
+    }
+
+    if (this.extension.source || this.extension.support) {
```

Comment:
Feels like these 2 should be separate

Reply:
I'll remove the support part but it isn't possible to move this since the data is in a different location in composer. Unless we want developers to specify the source a second time in composer.json

Reply:
Hmm I'm still not a fan of the actual LinkButton computation being repeated twice. Could we:

- Start by getting an array/object of the infoFields stuff
- Then, add support/source to that array/object
- Then, run the loop at line 196 over that array/object?